src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -2219,6 +2221,108 @@ def process_gallery_image_version_namespace(cmd, namespace):
 
         namespace.target_regions = regions_info
 
+    if hasattr(namespace, 'target_edge_zones') and namespace.target_edge_zones:
+        if 'none' in [zone.lower() for zone in namespace.target_edge_zones]:
+            namespace.target_edge_zones = []
+            return
```

Comment:
It is recommended to judge only the `namespace.target_edge_zones == 'None'` condition